apps/studio/src/features/dashboard/components/FolderSettingsModal/FolderSettingsModal.tsx

```diff
@@ -116,79 +124,75 @@ const SuspendableModalContent = ({
   }, [getFieldState, setValue, title])
 
   return (
-    <Suspense fallback={<Skeleton />}>
-      <ModalContent key={String(isOpen)}>
-        <form onSubmit={onSubmit}>
-          <ModalHeader>Edit "{originalTitle}"</ModalHeader>
-          <ModalCloseButton size="sm" />
-          <ModalBody>
-            <VStack alignItems="flex-start" spacing="1.5rem">
-              <FormControl isInvalid={!!errors.title}>
-                <FormLabel color="base.content.strong">
-                  Folder name
-                  <FormHelperText color="base.content.default">
-                    This will be the title of the index page of your folder.
-                  </FormHelperText>
-                </FormLabel>
-
-                <Input
-                  placeholder="This is a title for your new folder"
-                  {...register("title")}
-                />
-                {errors.title?.message ? (
-                  <FormErrorMessage>{errors.title.message}</FormErrorMessage>
-                ) : (
-                  <FormHelperText mt="0.5rem" color="base.content.medium">
-                    {MAX_FOLDER_TITLE_LENGTH - (title || "").length} characters
-                    left
-                  </FormHelperText>
-                )}
-              </FormControl>
-              <FormControl isInvalid={!!errors.permalink}>
-                <FormLabel color="base.content.strong">
-                  Folder URL
-                  <FormHelperText color="base.content.default">
-                    This will be applied to every child under this folder.
-                  </FormHelperText>
-                </FormLabel>
-                <Input
-                  placeholder="This is a url for your new page"
-                  {...register("permalink")}
-                />
-                {errors.permalink?.message && (
-                  <FormErrorMessage>
-                    {errors.permalink.message}
-                  </FormErrorMessage>
-                )}
-
-                <Box
-                  mt="0.5rem"
-                  py="0.5rem"
-                  px="0.75rem"
-                  bg="interaction.support.disabled"
-                >
-                  <Icon mr="0.5rem" as={BiLink} />
-                  {permalink}
-                </Box>
+    <ModalContent key={String(!!folderId)}>
```

Comment:
actually no need `key` to force a rerender here, since the parent will rerender whenever the folderId exists (the `!!folderId` conditional).